src/Agoda.Analyzers/AgodaCustom/AG0039MethodLineLengthAnalyzer.cs

```diff
@@ -1,13 +1,22 @@
 ﻿using System.Collections.Immutable;
+using System.Resources;
 using Agoda.Analyzers.Helpers;
 using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.CSharp;
 using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Text;
+using System.Collections.Generic;
+using System.Data;
+using System.Linq;
 
 namespace Agoda.Analyzers.AgodaCustom
 {
     [DiagnosticAnalyzer(LanguageNames.CSharp)]
-    public class AG0039UndocumentedMemberAnalyzer : DiagnosticAnalyzer
+    public class AG0039MethodLineLengthAnalyzer : DiagnosticAnalyzer
     {
+        private const int MaxLines = 40; // Adjust the maximum allowed lines as needed.
```

Comment:
I haven't tried, but something there might be possibility to make this configurable:
https://www.mytechramblings.com/posts/configure-roslyn-analyzers-using-editorconfig/

Reply:
that's cool, i think we can default it to 40 then make it configurable.

Reply:
was a bit tricky, but i think i've done it